routing/record/validation.go

```diff
@@ -73,13 +73,19 @@ func (v Validator) IsSigned(k key.Key) (bool, error) {
 // verifies that the passed in record value is the PublicKey
 // that matches the passed in key.
 func ValidatePublicKeyRecord(k key.Key, val []byte) error {
-	keyparts := bytes.Split([]byte(k), []byte("/"))
-	if len(keyparts) < 3 {
-		return errors.New("invalid key")
+	if len(k) != 38 {
+		return errors.New("invalid public key record key")
```

Comment:
true... I guess i could switch to chopping off the prefix and checking that its a valid multihash